osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -279,14 +281,36 @@ protected override void UpdateMods()
             Mods.Value = client.LocalUser.Mods.Select(m => m.ToMod(ruleset)).Concat(SelectedItem.Value.RequiredMods).ToList();
         }
 
+        [Resolved(canBeNull: true)]
+        private DialogOverlay dialogOverlay { get; set; }
+
+        private bool exitConfirmed;
+
         public override bool OnBackButton()
         {
-            if (client.Room != null && settingsOverlay.State.Value == Visibility.Visible)
+            if (client.Room == null)
+            {
+                // room has not been created yet; exit immediately.
+                return base.OnBackButton();
+            }
+
+            if (settingsOverlay.State.Value == Visibility.Visible)
             {
                 settingsOverlay.Hide();
                 return true;
             }
 
+            if (!exitConfirmed && dialogOverlay != null)
+            {
+                dialogOverlay.Push(new ConfirmDialog("leave this multiplayer match", () =>
```

Comment:
This is not going to translate well. Maybe just split these out into separate dialogs?

Reply:
did consider this but it's all a bit pointless until we make a serious push for localisation (there are countless other similar cases). i guess we can just split out from now, sure.